include/crypto.h

```diff
@@ -55,6 +55,9 @@
 extern int32_t Crypto_Config_CryptoLib(uint8_t sadb_type, uint8_t cryptography_type, uint8_t crypto_create_fecf, uint8_t process_sdls_pdus,
                                        uint8_t has_pus_hdr, uint8_t ignore_sa_state, uint8_t ignore_anti_replay,
                                        uint8_t unique_sa_per_mapid, uint8_t crypto_check_fecf, uint8_t vcid_bitmask, uint8_t crypto_increment_nontransmitted_iv);
+extern int32_t Crypto_Config_CryptoLib2(uint8_t sadb_type, uint8_t cryptography_type, uint8_t crypto_create_fecf, uint8_t process_sdls_pdus,
+                                       uint8_t has_pus_hdr, uint8_t ignore_sa_state, uint8_t ignore_anti_replay,
+                                       uint8_t unique_sa_per_mapid, uint8_t crypto_check_fecf, uint8_t vcid_bitmask, uint8_t crypto_increment_nontransmitted_iv, uint8_t padding_control);                                       
```

Comment:
Configuration functions are how CryptoLib library users interact with and configure the library to behave in meaningful ways for their setup... Each configuration should answer the question "What control/functionality does this add that users need?" -- EG, sadb_type allows users to connect to a MariaDB data store, or an inmemory data store -- something different users will want.

What control/functionality does this new "padding_control" configuration option add that users need? I think the answer is actually "none". "padding_control" appears to be mostly unused. The only place it somewhat matters is in the kmc_crypto interface (in which case it can be assumed as a transient property), but it's not even really necessary there since that's the only way that cryptography interface behaves. Delete this entire "Crypto_Config_CryptoLib2" function -- it's unnecessary. Padding_control behaves in very specific ways for each cryptography interface implementation -- we don't need users configuring anything here.

Reply:
That was my method for removing the padding within KMC.  We might be able to do it differently.  But I needed some way to pass into the KMC template that there was padding that needed to be removed because it was not playing well.

If you have another way to pass this in, and make things happy.  I'll be glad to change it.

Reply:
Sure, just change the if condition:
`*ecs == CRYPTO_CIPHER_AES256_CBC && crypto_config->padding_control == KMC_PADDING`
to
`*ecs == CRYPTO_CIPHER_AES256_CBC`
-- if you need to make sure the padding length is greater than 0 , it can be:
`*ecs == CRYPTO_CIPHER_AES256_CBC && padding > 0`

Reply:
OK, I see what you're saying.  I had to open the actual code.  Comments interleaved  in was making it difficult to keep track of what we were discussing.

Reply:
Fixed